shared_libraries/orca_shared/recovery/test/unit_tests/test_shared_recovery.py

```diff
@@ -157,88 +149,110 @@ def test_create_status_for_job_no_errors(self):
         {"AWS_REGION": "us-west-2"},
         clear=True,
     )
-    def test_update_status_for_file_no_errors(self):
+    def test_update_status_for_file_no_errors_status_success(self):
```

Comment:
Not really sure why this was broken into 4 tests seems like a lot of overhead to manage when the input and output is always expected to be the same with the exception of and ERROR status. At the most I could see 2 tests here but the 1 test we had before to cycle through the values seems more efficient and easy to maintain. Is there a good reason for the breakout?

Reply:
Unfortunately FIFO queue does not work with subtests. If we use the same input for different tests using  the four statuses, it throws an error. 

```
 IndexError: list index out of range
```

My assumption is since we are sending the same input to FIFO queue, it only accepts the first message and ignores the one ones.

Reply:
See comments from previous comment. We should be able to use subtests with fifo queue, we just need to add a delete message before the loop goes around again.

```python
self.fifo_queue.delete_messages(
                    Entries=[
                        {
                            "ReceiptHandle": queue_contents[0].receipt_handle,
                            "Id": queue_contents[0].message_id,
                        }
                    ]
                )
```

Reply:
Yup this works!